internal/cluster/cluster_test.go

```diff
@@ -2081,10 +2092,11 @@ var _ = Describe("SetVipsData", func() {
 			}
 			err := capi.SetVipsData(ctx, &cluster, t.apiVip, t.ingressVip, t.clusterApiLease, t.clusterIngressLease, db)
 			Expect(err != nil).To(Equal(t.errorExpected))
-			var c common.Cluster
-			Expect(db.Take(&c, "id = ?", clusterId.String()).Error).ToNot(HaveOccurred())
+			c := getClusterFromDB(clusterId, db)
 			Expect(c.APIVip).To(Equal(t.expectedApiVip))
+			Expect(network.GetApiVipById(&c, 0)).To(Equal(t.expectedApiVip))
 			Expect(c.IngressVip).To(Equal(t.expectedIngressVip))
+			Expect(network.GetIngressVipById(&c, 0)).To(Equal(t.expectedIngressVip))
```

Comment:
For the sake of completeness, could you please check here also for `len(c.APIVips)` and `len(c.IngressVips)`? This will make us sure that any deletion we perform on the DB is successful

Reply:
Sure

Reply:
Done